##### Importamos librerias

In [13]:
# Tratamiento de datos
import pandas as pd

# Exploración de archivos
import sys
import os

# Funciones personalizadas
sys.path.append(os.path.abspath('../src'))
import sp_eda as se

# Visualizaciones
import matplotlib.pyplot as plt
import seaborn as sns

##### Para mostrar todas las columnas y las filas

In [14]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

##### Cargamos el conjunto de datos desde un archivo XLSX (detalles de consumidores)

In [15]:
# El archivo tiene 3 hojas, las cargamos una por una
df_2012 = pd.read_excel('../data/raw/customer-details.xlsx', sheet_name = "2012", index_col=0)
df_2013 = pd.read_excel('../data/raw/customer-details.xlsx', sheet_name = "2013", index_col=0)
df_2014 = pd.read_excel('../data/raw/customer-details.xlsx', sheet_name = "2014", index_col=0)

In [16]:
df_2012.head()

,Income,Kidhome,Teenhome,Dt_Customer,NumWebVisitsMonth,ID
0,161770,1,0,2012-04-04,29,089b39d8-e4d0-461b-87d4-814d71e0e079
1,85477,1,1,2012-12-30,7,e9d37224-cb6f-4942-98d7-46672963d097
2,147233,1,1,2012-02-02,5,3f9f49b5-e410-4948-bf6e-f9244f04918b
3,121393,1,2,2012-12-21,29,9991fafb-4447-451a-8be2-b0df6098d13e
4,63164,1,2,2012-06-20,20,eca60b76-70b6-4077-80ba-bc52e8ebb0eb


In [17]:
df_2013.head()

,Income,Kidhome,Teenhome,Dt_Customer,NumWebVisitsMonth,ID
0,82407,0,1,2013-10-07,29,ef81336c-e41a-46d0-8a30-5d4ac3b836be
1,24877,0,0,2013-04-21,32,50b224ac-84b4-42a7-aba9-3e1d1d620479
2,78839,1,0,2013-11-25,21,9cb15c53-b647-464a-add9-09cdae04897c
3,113859,1,1,2013-01-06,12,f8aaf334-949d-4943-a214-b64fe9c088b4
4,147317,0,0,2013-07-10,14,885fe85a-3718-4760-a32a-a57628cf7528


In [18]:
df_2014.head()

,Income,Kidhome,Teenhome,Dt_Customer,NumWebVisitsMonth,ID
0,168812,1,1,2014-06-10,27,4d6c811c-0f24-4a73-b374-2ebd1f80d46e
1,108615,2,1,2014-04-05,24,eef0f24f-cbdb-4c67-b371-09d7bc59695e
2,179041,0,2,2014-09-22,4,bbfdcedd-199d-4927-94bb-5dad051ff471
3,116639,2,2,2014-08-16,3,e8ebb67d-4ee1-4615-b701-eefe3af607eb
4,49537,0,1,2014-09-05,28,f7f79ace-0024-4505-a401-f340946f0942


##### 

In [19]:
# Concatenamos tres hojas XSLX en un único DataFrame
df_customers = pd.concat([df_2012, df_2013, df_2014])

## Analisis preliminar df_customers

In [20]:
se.analisis_rapido(df_customers)

LAS 5 PRIMERAS COLUMNAS DE DATAFRAME SON:


,Income,Kidhome,Teenhome,Dt_Customer,NumWebVisitsMonth,ID
0,161770,1,0,2012-04-04,29,089b39d8-e4d0-461b-87d4-814d71e0e079
1,85477,1,1,2012-12-30,7,e9d37224-cb6f-4942-98d7-46672963d097
2,147233,1,1,2012-02-02,5,3f9f49b5-e410-4948-bf6e-f9244f04918b
3,121393,1,2,2012-12-21,29,9991fafb-4447-451a-8be2-b0df6098d13e
4,63164,1,2,2012-06-20,20,eca60b76-70b6-4077-80ba-bc52e8ebb0eb


INFORMACIÓN BÁSICA DE DATAFRAME:
<class 'pandas.DataFrame'>
Index: 43170 entries, 0 to 14089
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Income             43170 non-null  int64         
 1   Kidhome            43170 non-null  int64         
 2   Teenhome           43170 non-null  int64         
 3   Dt_Customer        43170 non-null  datetime64[us]
 4   NumWebVisitsMonth  43170 non-null  int64         
 5   ID                 43170 non-null  str           
dtypes: datetime64[us](1), int64(4), str(1)
memory usage: 2.3 MB


None

EL NÚMERO DE DUPLICADOS ES: 0
EL NÚMERO DE VALORES NULOS ES:


Income               0
Kidhome              0
Teenhome             0
Dt_Customer          0
NumWebVisitsMonth    0
ID                   0
dtype: int64

El dataset df_customers cuenta con 
43170 filas, donde cada una representa información demográfica y de comportamiento de compra de los clientes del banco y 6 columnas, que incluyen cuatro variables numéricas, una columna de fecha y una categórica.

Columnas del dataset:

●	Income: Representa el ingreso anual del cliente en términos monetarios.

●	Kidhome: Indica el número de niños en el hogar del cliente.

●	Teenhome: Indica el número de adolescentes en el hogar del cliente.

●	Dt_Customer: Representa la fecha en que el cliente se convirtió en cliente de la empresa.

●	NumWebVisitsMonth: Indica la cantidad de visitas mensuales del cliente al sitio web de la empresa.

●	ID: Identificador único del cliente.

El conjunto de datos **NO** tiene ni duplicados ni valores nulos.

##### Columnas numéricas:

In [21]:
se.descriptivo_num(df_customers)

Variables numéricas son:
 Index(['Income', 'Kidhome', 'Teenhome', 'Dt_Customer', 'NumWebVisitsMonth'], dtype='str')

Las estadísticas básicas:


,count,mean,min,25%,50%,75%,max,std
Income,43170.0,93227.39,5841.0,49608.0,93009.5,136740.5,180802.0,50498.18
Kidhome,43170.0,1.0,0.0,0.0,1.0,2.0,2.0,0.82
Teenhome,43170.0,1.0,0.0,0.0,1.0,2.0,2.0,0.82
Dt_Customer,43170,2013-05-11 22:09:47.407922,2012-01-01 00:00:00,2012-07-16 00:00:00,2013-03-01 00:00:00,2014-03-28 00:00:00,2014-12-31 00:00:00,NaN
NumWebVisitsMonth,43170.0,16.59,1.0,9.0,17.0,25.0,32.0,9.24


##### Descripción de los datos numéricos

Observación preliminar de los datos numericos:

El ingreso anual presenta una media cercana a 93.000, con una alta dispersión, lo que indica una gran variabilidad entre clientes.

La mayoría de los hogares tiene uno o dos niños o adolescentes, con valores medios cercanos a 1.

La variable Dt_Customer recoge la fecha de alta de los clientes, con registros comprendidos entre 01.01.2012 y 31.12.2014, lo que permite realizar análisis temporales de antigüedad.

El número de visitas mensuales al sitio web es relativamente alto, con una media de alrededor de 17 visitas, aunque se observan diferencias importantes entre clientes.

##### Columnas categóricas:

In [ ]:
se.descriptivo_cat(df_customers)

Variables categoricas son:
 Index(['ID'], dtype='str')

Las estadísticas básicas:


,count,unique,top,freq
ID,43170,43170,089b39d8-e4d0-461b-87d4-814d71e0e079,1


La columna *ID* tiene 43170 valores unicos:
Los 10 valores con más frequencia son:


ID
089b39d8-e4d0-461b-87d4-814d71e0e079    1
e9d37224-cb6f-4942-98d7-46672963d097    1
3f9f49b5-e410-4948-bf6e-f9244f04918b    1
9991fafb-4447-451a-8be2-b0df6098d13e    1
eca60b76-70b6-4077-80ba-bc52e8ebb0eb    1
d63ede72-0b6d-45b1-8872-385ac6897f65    1
5e3483e5-236d-437d-8351-541f9d09b9dd    1
87fdc08b-30ae-4dab-803f-561ecdf27ff0    1
87b79988-2be5-419d-88f4-56655852c565    1
ea6b7d04-9271-4c0a-a01f-07795d164aba    1
Name: count, dtype: int64

Observación preliminar de los datos categóricos:

La variable ID actúa como identificador único, ya que todos sus valores son distintos.

##### Guardamos  dataset de clientes df_customers transformado:

In [ ]:
df_customers.to_csv('../data/processed/consum_limpio.csv', index=False)